pkg/controller/elasticsearch/driver/upgrade_predicates_test.go

```diff
@@ -675,48 +676,50 @@ func TestDeletionStrategy_SortFunction(t *testing.T) {
 			name: "Mixed nodes",
 			fields: fields{
 				upgradeTestPods: newUpgradeTestPods(
-					newTestPod("masters-0").needsUpgrade(true),
-					newTestPod("data-0").needsUpgrade(true),
-					newTestPod("masters-1").needsUpgrade(true),
-					newTestPod("data-1").needsUpgrade(true),
-					newTestPod("masters-2").needsUpgrade(true),
+					// use "amasters" rather than "masters" to ensure we are not relying on the name sort accidentally
+					newTestPod("amasters-0").isMaster(true).needsUpgrade(true),
+					newTestPod("data-0").isData(true).needsUpgrade(true),
+					newTestPod("amasters-1").isMaster(true).needsUpgrade(true),
+					newTestPod("data-1").isData(true).needsUpgrade(true),
+					newTestPod("amasters-2").isMaster(true).needsUpgrade(true),
```

Comment:
Could we add a pod here from another master sset with `masters-0` name? Just to be sure we cover both cases.